src/movies.js

```diff
@@ -1,13 +1,90 @@
 // Iteration 1: All directors? - Get the array of all directors.
+//console.log(movies);
+
+// function getAllDirectors(movies){
+//     return movies.map(function (movie)
+//     {
+//     return movie.director;
+// })
+// }
+
+//easier ES6 notation
+function getAllDirectors(movies){
+    return movies.map(movie => movie.director);
+}
+console.log(getAllDirectors(movies));
+console.clear();
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+// function howManyMovies(movies){
+//     let filteredMovies = movies.filter(function (movie){
+//         if (movie.director === 'Steven Spielberg') {
+//             return true
+//         }
+//         return filteredMovies;
+        
+//         }
+
+//     )}
+
+//     console.log(howManyMovies(movies));
+function howManyMovies(movies){
+
+    let filteredMovies = movies.filter(function(movie){
+        if ((movie.director === 'Steven Spielberg') && (movie.genre.includes('Drama'))){
+            return true}
+            
+    })
+    return filteredMovies.length;
+} 
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+ function ratesAverage(movies){
+     if (movies.length === 0){return 0
+    } else {
+        let avg = movies.reduce(function(acc, val){
+            //conditioanl statement, if not return acc.
+            if (val ===''){
+                return 0;
+            } else {
+                return acc + val.rate;
+            }
+        },0)/movies.length;
+        return parseFloat(avg.toFixed(2));
+    }
+}
```

Comment:
"val" here is a movie object, it could be totally empty or it could be only missing the rate. If you want to check whether the rate field is empty, you could do val.rate === "" but a better way that would work even if the whole movie is empty is to check `if (!val.rate) `. It is like checking if val.rate does not exist. In this case, you would still need to return the `acc` but without adding anything to it.

 ```
       let avg = movies.reduce(function(acc, val){
            //conditional statement, if not return acc.
            if (!val.rate){
                return acc;
            } else {
                return acc + val.rate;
            }
        },0)/movies.length;